### ISSUE TITLE
Found in #bug-reports: Deleting a prepared spell via Harvest Scythe crashes your game next time you open the spellbook or engage in battle

### ISSUE THREAD

**Issue by ncrecc** [TerryCavanagh/diceydungeons.com#1382]

**Comment by Steins-fr** [comment]
Just done that during cornelius battle. I am waiting a patch to continu my save.
I consumed Harvest Scythe that was in prepared slot.

Logs crash "enter battle":
```
--------------------------------------
filename:	diceydungeons
package:	com.terrycavanaghgames.diceydungeons
version:	v1.3 / 0.11.0 (mod API)
sess. ID:	dicey_dungeons_2019-09-01_22'16'48
started:	2019-09-01 22:16:48
--------------------------------------
crashed:	2019-09-01 22:17:00
duration:	00:00:12
error:		ERROR in callscenemethod(Combat,update) static : Null Object Reference, stack = 
Called from elements.Equipment.render (elements/Equipment.hx line 1648)
Called from elements.Equipment.draw (elements/Equipment.hx line 1610)
Called from states.Combat.update (states/Combat.hx line 529)
Called from Reflect.callMethod (/usr/local/lib/haxe/std/cpp/_std/Reflect.hx line 55)
stack:
haxegon.Scene.callscenemethod (haxegon/Scene.hx line 93)
haxegon.Scene.update (haxegon/Scene.hx line 53)
haxegon.Core.doupdate (haxegon/Core.hx line 268)
haxegon.Core.onEnterFrame (haxegon/Core.hx line 182)
starling.events.EventDispatcher.__invokeEvent (starling/events/EventDispatcher.hx line 184)
starling.events.EventDispatcher.dispatchEvent (starling/events/EventDispatcher.hx line 144)
starling.display.DisplayObject.dispatchEvent (starling/display/DisplayObject.hx line 778)
starling.display.DisplayObjectContainer.broadcastEvent (starling/display/DisplayObjectContainer.hx line 449)
starling.display.Stage.advanceTime (starling/display/Stage.hx line 108)
starling.core.Starling.advanceTime (starling/core/Starling.hx line 469)
starling.core.Starling.nextFrame (starling/core/Starling.hx line 455)
starling.core.Starling.onEnterFrame (starling/core/Starling.hx line 682)
openfl.events.EventDispatcher.__dispatchEvent (openfl/events/EventDispatcher.hx line 443)
openfl.display.DisplayObject.__dispatch (openfl/display/DisplayObject.hx line 1236)
```

Log crash "check spellbook":
```
--------------------------------------
filename:	diceydungeons
package:	com.terrycavanaghgames.diceydungeons
version:	v1.3 / 0.11.0 (mod API)
sess. ID:	dicey_dungeons_2019-09-01_19'16'18
started:	2019-09-01 19:16:18
--------------------------------------
crashed:	2019-09-01 22:16:14
duration:	02:59:56
error:		ERROR in callscenemethod(ViewSpellbook,update) static : Error: There's no spell template for ""., stack = 
Called from Gamedata.getspelltemplate (Gamedata.hx line 830)
Called from elements.Spellbook.createspell (elements/Spellbook.hx line 90)
Called from states.ViewSpellbook.changeprecast (states/ViewSpellbook.hx line 135)
Called from states.ViewSpellbook.update (states/ViewSpellbook.hx line 378)
Called from Reflect.callMethod (/usr/local/lib/haxe/std/cpp/_std/Reflect.hx line 55)
stack:
haxegon.Scene.callscenemethod (haxegon/Scene.hx line 93)
haxegon.Scene.update (haxegon/Scene.hx line 53)
haxegon.Core.doupdate (haxegon/Core.hx line 268)
haxegon.Core.onEnterFrame (haxegon/Core.hx line 182)
starling.events.EventDispatcher.__invokeEvent (starling/events/EventDispatcher.hx line 184)
starling.events.EventDispatcher.dispatchEvent (starling/events/EventDispatcher.hx line 144)
starling.display.DisplayObject.dispatchEvent (starling/display/DisplayObject.hx line 778)
starling.display.DisplayObjectContainer.broadcastEvent (starling/display/DisplayObjectContainer.hx line 449)
starling.display.Stage.advanceTime (starling/display/Stage.hx line 108)
starling.core.Starling.advanceTime (starling/core/Starling.hx line 469)
starling.core.Starling.nextFrame (starling/core/Starling.hx line 455)
starling.core.Starling.onEnterFrame (starling/core/Starling.hx line 682)
openfl.events.EventDispatcher.__dispatchEvent (openfl/events/EventDispatcher.hx line 443)
openfl.display.DisplayObject.__dispatch (openfl/display/DisplayObject.hx line 1236)
```

**Comment by Yoco64** [comment]
I have exactly the same problem, in 2 runs on Episode 5 and 6.
In the Episode 5, i could not open the Spellbook, but i could launch the Boss battle and finish the run.
In the Episode 6, i can't finish the run because i can't open the spellbook or launch battle, with the same error message that you.

**Comment by Steins-fr** [comment]
If you have a shop and enough money, you can buy a spell to replace the spell consumed. It worked for me. Otherwise, you can't finish the run.

**Comment by hunky-dory** [comment]
I had this issue in a Bonus Round run, where the game would crash even if I tried to buy a spell from the shop (I had the pirate modifier enabled, so Harvest Scythe had removed two spells)

The errors generated vary depending on where you cause the spellbook to be invoked (eg buying something from a shop, clicking spellbook from the overworld screen, or entering a fight) but I got this one when loading in to the boss battle:
```
CrashDumper.hx:248: CRASH session.id = dicey_dungeons_2019-09-11_19'02'37
CrashDumper.hx:249: MESSAGE = --------------------------------------
filename:	diceydungeons
package:	com.terrycavanaghgames.diceydungeons
version:	v1.3 / 0.11.0 (mod API)
sess. ID:	dicey_dungeons_2019-09-11_19'02'37
started:	2019-09-11 19:02:37
--------------------------------------
crashed:	2019-09-11 19:02:51
duration:	00:00:14
error:		ERROR in callscenemethod(ViewSpellbook,update) static : Error: There's no spell template for ""., stack = 
Called from Gamedata.getspelltemplate (Gamedata.hx line 830)
Called from elements.Spellbook.createspell (elements/Spellbook.hx line 90)
Called from states.ViewSpellbook.changeprecast (states/ViewSpellbook.hx line 135)
Called from states.ViewSpellbook.update (states/ViewSpellbook.hx line 384)
Called from Reflect.callMethod (/usr/local/lib/haxe/std/cpp/_std/Reflect.hx line 55)
stack:
haxegon.Scene.callscenemethod (haxegon/Scene.hx line 93)
haxegon.Scene.update (haxegon/Scene.hx line 53)
haxegon.Core.doupdate (haxegon/Core.hx line 268)
haxegon.Core.onEnterFrame (haxegon/Core.hx line 182)
starling.events.EventDispatcher.__invokeEvent (starling/events/EventDispatcher.hx line 184)
starling.events.EventDispatcher.dispatchEvent (starling/events/EventDispatcher.hx line 144)
starling.display.DisplayObject.dispatchEvent (starling/display/DisplayObject.hx line 778)
starling.display.DisplayObjectContainer.broadcastEvent (starling/display/DisplayObjectContainer.hx line 449)
starling.display.Stage.advanceTime (starling/display/Stage.hx line 108)
starling.core.Starling.advanceTime (starling/core/Starling.hx line 469)
starling.core.Starling.nextFrame (starling/core/Starling.hx line 455)
starling.core.Starling.onEnterFrame (starling/core/Starling.hx line 682)
openfl.events.EventDispatcher.__dispatchEvent (openfl/events/EventDispatcher.hx line 443)
openfl.display.DisplayObject.__dispatch (openfl/display/DisplayObject.hx line 1236)


CrashDumper.hx:142: onError(EOF)
```

**Comment by TerryCavanagh** [comment]
This one is fixed for v1.5. Thank you for the bug reports!
